malariagen_data/anoph/snp_data.py

```diff
@@ -834,6 +846,18 @@ def _locate_site_class(
             else:
                 raise NotImplementedError(site_class)
 
+            # N.B., site annotations data are provided for every position in the genome. We need to
+            # therefore subset to SNP positions.
+            pos = self.snp_sites(
+                region=region,
+                field="POS",
+                site_mask=site_mask,
+                inline_array=inline_array,
+                chunks=chunks,
+            )
+            idx = (pos - 1).compute()
+            loc_ann = da.take(loc_ann, idx, axis=0)
```

Comment:
This is a key move, we delay the indexing of the site annotations data to SNP positions until later on. This avoids the indexing array (idx) getting copied multiple times.